### ISSUE TITLE
[FEATURE] Leverage/Collaborate with Karaoke Nerds

### ISSUE THREAD

**Issue by Deastrom** [vicwomg/pikaraoke#310]
**Is your feature request related to a problem? Please describe.**
There is a little bit of frustration with the wide open <search> + "karaoke" youtube search and finding quality karaoke videos.

**Describe the solution you'd like**
https://www.karaokenerds.com/About/

Karaoke Nerds has some functionality that could greatly improve the video search experience in the application. A strong integration could also lead to leveraging Karaoke Nerds user experience features as well, allowing pikaraoke to focus on downloading and queuing up songs.

**Describe alternatives you've considered**
I've already kind of gone down this rabbit hole but the code for karaoke nerds isn't open source AND it's not leveraging any kind of backend api. So this integration would likely be more like web parsing, which isn't ideal.

I don't know if it's possible to iframe karaokenerds and add a "download and add to queue" or "add to queue"(if already downloaded) button in the iframe (i don't front end much, or at all)

**Additional context**
Love the project. Used it at my new years eve party last night with the latest improvements. Keep up the good work.

**Comment by RecursiveFunctions** [comment]
I and a few others have considered it. I'll work on it in a separate branch. Not sure how I'd go about it though. Might need to use a scraper as suggested. Maybe have a manual 'update' button on the settings webpage for it too. 

One thing to keep in mind is that karaoke nerds is not all-inclusive of all the karaoke videos on youtube. It doesn't include midi vids, and it's not always up to date, and some channels like sing king are not included.

**Comment by Deastrom** [comment]
Yeah, the crux of the problem is video/karaoke quality. I suppose I'd be really happy with a "priority channels" list or something like that. But I know that'd make the youtube search more complicated.

**Comment by Deastrom** [comment]
> I and a few others have considered it. I'll work on it in a separate branch. Not sure how I'd go about it though. Might need to use a scraper as suggested. Maybe have a manual 'update' button on the settings webpage for it too.
> 
> One thing to keep in mind is that karaoke nerds is not all-inclusive of all the karaoke videos on youtube. It doesn't include midi vids, and it's not always up to date, and some channels like sing king are not included.

Also, if you DM him in discord, i'm sure he'd work with you

**Comment by Deastrom** [comment]
[My Fork](https://github.com/vicwomg/pikaraoke/compare/master...Deastrom:pikaraokenerds:master)

came up with this for this integration. It parses karaokerds.com and also leverages yt-dlp to crawl a couple of well known karaoke channels.

**Comment by vicwomg** [comment]
@Deastrom  Tried the fork and it would not start up because there is an incompatibility with `httpx` and gevent. 

`NotImplementedError: unsupported platform or primitives trio depends on are monkey-patched out by gevent`

My suggestion (which seemed to work locally) is to remove the httpx dependency and use the python standard library's `requests.get` instead.

Other suggestions:
- The sync takes a very long time and is resource-intensive. Why not check in a zipped version of the json index file (compresses down to < 2mb) into a resources folder and extract that on first time launch if no existing file is there? This file could be updated on every release. Or I even wonder if it could be hosted on a centralized server that updates regularly, which would spare anyone the need to do this manually.
- The "play" button is a bit misleading, it opens youtube instead of queuing the song. Suggest using the external link icon: https://fontawesome.com/v4/icon/external-link
- If you'd like us to consider merging this to the main pikaraoke project, please don't replace the Pikaraoke logo.

**Comment by Deastrom** [comment]
suggested modifications made. although i'm not sure you'd like this upstream or not. at least not without unifying the styling.

**Comment by vicwomg** [comment]
> I did not add the uploaded cache. Ideally the refresh only needs to be run occasionally, like just before a party. Uploading a zipped copy of the list is like uploading an initial database and that doesn't seem quite normal for a git project.

Not abnormal at all. It's essentially database seed file. Many databases need to be seeded in some way before they can be effectively used. 

I think having a "curated" list of quality tracks definitely a worthy contribution to the upstream project. I'll have a look at how integrate it in a way that makes sense. The most transparent way I would think is to simply have some starred items that show up in the search box that pull from this list.
